napi_from_js.md

```diff
@@ -0,0 +1,367 @@
+
+// Convert the result back to a native type
+int32_t result;
+status = napi_get_value_int32(env, return_val, &result);
+if (status != napi_ok) return;
+```
+
+### *napi_create_function*
+
+#### Signature
+```C
+napi_status napi_create_function(napi_env env,
+                                 const char* utf8name,
+                                 napi_callback cb,
+                                 void* data,
+                                 napi_value* result);
+```
+
+#### Parameters
+- `[in]  env `: The environment that the API is invoked under
+- `[in]  utf8Name`: The name of the function, encoded as UTF8. This is visible
+within JavaScript as the new function object's `name` property.
+- `[in]  cb`: The native function which should be called when this function 
+object is invoked
+- `[in]  data`: User-provided data context. This will be passed back into the 
+function when invoked later.
+- `[out] result`: `napi_value` representing the JavaScript function object for
+the newly created function
+
+#### Description
+This API allows an add-on author to create a function object in native code. 
+This is the primary mechanism to allow calling *into* the add-on's native code
+*from* Javascript.
+
+Note however, the newly created function is not automatically  visible from
+script after this call. Instead, you must explicitly set a property on the
+Global object (or any other object that is visible to JavaScript), in order for
+the function to be accessible from script. For instance, if you wish to expose a
+function as part of the add-on's module exports, you can set the newly created
+function on the exports object. A sample module might look as follows:
+```C
+void SayHello(napi_env env, napi_callback_info info) {
+  printf("Hello\n");
+}
+
+void Init(napi_env env, napi_value exports, napi_value module, void* priv) {
+  napi_status status;
+
+  napi_value fn;
+  status =  napi_create_function(env, NULL, SayHello, NULL, &fn);
+  if (status != napi_ok) return;
+
+  status = napi_set_named_property(env, exports, "sayHello", fn);
+  if (status != napi_ok) return;
+}
+
+NAPI_MODULE(add-on, Init)
+```
+
+Given the above code, the add-on can be used from JavaScript as follows:
+```js
+var myaddon = require('myadd-on');
+myaddon.sayHello();
+```
+
+### *napi_get_cb_args*
+
+#### Signature
+```C
+napi_status napi_get_cb_args(napi_env env,
+                             napi_callback_info cbinfo,
+                             napi_value* buffer,
+                             size_t bufferlength)
+```
+
+#### Parameters
+- `[in] env`: The environment that the API is invoked under
+- `[in] cbinfo`: The callback info passed into the callback function
+- `[in] buffer`: Array of `napi_value` into which the arguments to the 
+callback function will be copied. This array must be user-allocated
+and have space to store at least `bufferlength`-count elements. Any
+additional space will be filled with `napi_value` values that represent 
+`undefined`
```

Comment:
Period after &#96;undefined&#96;.
Also, add the converse: "Conversely, if &#96;bufferlength&#96; is insufficient to account for all the arguments that were passed from JavaScript, only the first &#96;bufferlength&#96; arguments will be copied into &#96;buffer&#96;."